packages/asset-service/src/generateAssetData/ethereum/idleVaults.ts

```diff
@@ -20,44 +32,55 @@ const explorerData = {
 const getIdleVaults = async (): Promise<Asset[]> => {
   const vaults: IdleVault[] = await idleSdk.getVaults()
 
-  return vaults.map((vault: IdleVault) => {
+  const assets: Asset[] = []
+
+  for (const vault of vaults) {
     const assetId = toAssetId({ chainId, assetNamespace: 'erc20', assetReference: vault.address })
     const displayIcon = `https://raw.githubusercontent.com/Idle-Labs/idle-dashboard/master/public/images/tokens/${vault.tokenName}.svg`
-    return {
+    const symbol = (await getAssetSymbol(vault.address)) ?? vault.tokenName // Alchemy XHRs might fail, you never know
+
+    assets.push({
       color: colorMap[assetId] ?? '#FFFFFF',
       icon: displayIcon,
       name: vault.poolName,
       precision: Number(18),
-      symbol: vault.tokenName,
-      tokenId: toLower(vault.address),
```

Comment:
We don't use this property, this is yearn-specific and was added because monkey see monkey do